cfme/tests/infrastructure/test_esx_direct_host.py

```diff
@@ -13,14 +13,16 @@
 
 
 def pytest_generate_tests(metafunc):
-    argnames, argvalues, idlist = testgen.provider_by_type(metafunc, ['virtualcenter'])
+    argnames, argvalues, idlist = testgen.providers_by_type(metafunc, ['virtualcenter'])
     argnames = argnames + ["_host_provider"]
 
     new_idlist = []
     new_argvalues = []
 
     for i, argvalue_tuple in enumerate(argvalues):
         args = dict(zip(argnames, argvalue_tuple))
+        # TODO
+        # All this should be replaced with a proper ProviderFilter passed to testgen.providers()
```

Comment:
So looking at this, if we ask for virtualcenter on line 16, this shouldn't be required, we should also add the required_fields so that lines 26 can go. Lines 32-> 35 can probably go with using required fields.